<?xml version="1.0" encoding="UTF-8" ?>
<testsuite errors="34" failures="2" hostname="cs-210-206.cs.ucla.edu" name="com.google.javascript.jscomp.CheckGlobalThisTest" skipped="0" tests="52" time="0.162" timestamp="2015-03-09T03:17:12">
  <properties>
    <property name="java.vendor" value="Oracle Corporation" />
    <property name="lib.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib" />
    <property name="refasterjs-jarfile" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/refasterjs.jar" />
    <property name="sun.java.launcher" value="SUN_STANDARD" />
    <property name="sun.management.compiler" value="HotSpot 64-Bit Tiered Compilers" />
    <property name="os.name" value="Mac OS X" />
    <property name="sun.boot.class.path" value="/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/resources.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/rt.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/sunrsasign.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/jsse.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/jce.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/charsets.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/jfr.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/classes" />
    <property name="maven-repository-id" value="sonatype-nexus-staging" />
    <property name="java.vm.specification.vendor" value="Oracle Corporation" />
    <property name="ant.home" value="//Users/linmanna/ant" />
    <property name="java.runtime.version" value="1.7.0_75-b13" />
    <property name="gen.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/gen" />
    <property name="user.name" value="linmanna" />
    <property name="ant.file.compiler" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build.xml" />
    <property name="build.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build" />
    <property name="ant.build.javac.target" value="1.7" />
    <property name="user.language" value="zh" />
    <property name="test.method" value="" />
    <property name="sun.boot.library.path" value="/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib" />
    <property name="ant.project.default-target" value="jar" />
    <property name="ant.project.name" value="compiler" />
    <property name="java.version" value="1.7.0_75" />
    <property name="user.timezone" value="" />
    <property name="sun.arch.data.model" value="64" />
    <property name="http.nonProxyHosts" value="local|*.local|169.254/16|*.169.254/16" />
    <property name="java.endorsed.dirs" value="/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/endorsed" />
    <property name="sun.cpu.isalist" value="" />
    <property name="sun.jnu.encoding" value="UTF-8" />
    <property name="file.encoding.pkg" value="sun.io" />
    <property name="file.separator" value="/" />
    <property name="java.specification.name" value="Java Platform API Specification" />
    <property name="protoc.executable" value="protoc" />
    <property name="classes.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/classes" />
    <property name="java.class.version" value="51.0" />
    <property name="testClasses.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/test" />
    <property name="user.country" value="CN" />
    <property name="java.home" value="/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre" />
    <property name="java.vm.info" value="mixed mode" />
    <property name="externs.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/externs" />
    <property name="ant.file" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build.xml" />
    <property name="os.version" value="10.9.5" />
    <property name="javac.debug" value="on" />
    <property name="path.separator" value=":" />
    <property name="java.vm.version" value="24.75-b04" />
    <property name="ant.library.dir" value="//Users/linmanna/ant/lib" />
    <property name="java.awt.printerjob" value="sun.lwawt.macosx.CPrinterJob" />
    <property name="webservice-jarfile" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/webservice.jar" />
    <property name="sun.io.unicode.encoding" value="UnicodeBig" />
    <property name="test.fork" value="true" />
    <property name="webservice.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/src/com/google/javascript/jscomp/webservice" />
    <property name="awt.toolkit" value="sun.lwawt.macosx.LWCToolkit" />
    <property name="buildlib.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/lib" />
    <property name="socksNonProxyHosts" value="local|*.local|169.254/16|*.169.254/16" />
    <property name="ftp.nonProxyHosts" value="local|*.local|169.254/16|*.169.254/16" />
    <property name="ant.build.javac.source" value="1.7" />
    <property name="refasterjs.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/src/com/google/javascript/refactoring" />
    <property name="user.home" value="/Users/linmanna" />
    <property name="java.specification.vendor" value="Oracle Corporation" />
    <property name="test.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/test" />
    <property name="java.library.path" value="/Users/linmanna/Library/Java/Extensions:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java:." />
    <property name="java.vendor.url" value="http://java.oracle.com/" />
    <property name="java.vm.vendor" value="Oracle Corporation" />
    <property name="gopherProxySet" value="false" />
    <property name="build.relVersion" value="00e3533" />
    <property name="java.runtime.name" value="Java(TM) SE Runtime Environment" />
    <property name="sun.java.command" value="org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner testsfile=/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/junittestcases4622356093930181862.properties skipNonTests=false filtertrace=true haltOnError=false haltOnFailure=false formatter=org.apache.tools.ant.taskdefs.optional.junit.SummaryJUnitResultFormatter showoutput=true outputtoformatters=true logfailedtests=true threadid=0 logtestlistenerevents=false formatter=org.apache.tools.ant.taskdefs.optional.junit.BriefJUnitResultFormatter formatter=org.apache.tools.ant.taskdefs.optional.junit.XMLJUnitResultFormatter,/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/testoutput/IGNORETHIS.xml crashfile=/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/junitvmwatcher882389895192408063.properties propsfile=/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/junit6374663990081372210.properties" />
    <property name="java.class.path" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/classes:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/ant-launcher.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/ant.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/args4j.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/gson.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/guava.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/jsr305.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/junit.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/protobuf-java.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/lib/truth.jar:/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/test:/Users/linmanna/ant/lib/ant-launcher.jar:/Users/linmanna/ant/lib/ant.jar:/Users/linmanna/ant/lib/ant-junit.jar:/Users/linmanna/ant/lib/ant-junit4.jar" />
    <property name="ant.version" value="Apache Ant(TM) version 1.9.4 compiled on April 29 2014" />
    <property name="test.class" value="*Test" />
    <property name="java.vm.specification.name" value="Java Virtual Machine Specification" />
    <property name="ant.file.type" value="file" />
    <property name="java.vm.specification.version" value="1.7" />
    <property name="sun.cpu.endian" value="little" />
    <property name="sun.os.patch.level" value="unknown" />
    <property name="java.io.tmpdir" value="/var/folders/_p/3swdt4h96dn79vyt21dtny9h0000gn/T/" />
    <property name="java.vendor.url.bug" value="http://bugreport.sun.com/bugreport/" />
    <property name="java.awt.graphicsenv" value="sun.awt.CGraphicsEnvironment" />
    <property name="maven-repository-url" value="https://oss.sonatype.org/service/local/staging/deploy/maven2/" />
    <property name="os.arch" value="x86_64" />
    <property name="webservice-classes.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/webservice-classes" />
    <property name="java.ext.dirs" value="/Users/linmanna/Library/Java/Extensions:/Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre/lib/ext:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java" />
    <property name="user.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler" />
    <property name="ant.file.type.compiler" value="file" />
    <property name="line.separator" value="&#xa;" />
    <property name="java.vm.name" value="Java HotSpot(TM) 64-Bit Server VM" />
    <property name="refasterjs-classes.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/refasterjs-classes" />
    <property name="basedir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler" />
    <property name="ant.java.version" value="1.7" />
    <property name="ant.core.lib" value="/Users/linmanna/ant/lib/ant.jar" />
    <property name="file.encoding" value="UTF-8" />
    <property name="closure.build.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build" />
    <property name="compiler-jarfile" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/compiler.jar" />
    <property name="java.specification.version" value="1.7" />
    <property name="src.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/src" />
    <property name="ant.project.invoked-targets" value="clean,compile,compile-tests,test" />
    <property name="javadoc.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/build/javadoc" />
    <property name="tools.dir" value="/Users/linmanna/Documents/proj/UCLA-CS-239-Winter-2015/Part_B/Benchmark/closure-compiler/tools" />
  </properties>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testGlobalThis1" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testGlobalThis2" time="0.003" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testGlobalThis3" time="0.003" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testGlobalThis4" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testGlobalThis5" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testGlobalThis6" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testGlobalThis7" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction1" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION a): testcode:1:0&#xa;function a() { return this; }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function a() { return this; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION a): testcode:1:0
function a() { return this; }
  Parent(SCRIPT): testcode:1:0
function a() { return this; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction1(CheckGlobalThisTest.java:68)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction2" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION a): testcode:1:0&#xa;function a() { this.complex = 5; }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function a() { this.complex = 5; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION a): testcode:1:0
function a() { this.complex = 5; }
  Parent(SCRIPT): testcode:1:0
function a() { this.complex = 5; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction2(CheckGlobalThisTest.java:72)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction3" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:8&#xa;var a = function() { return this; }&#xa;  Parent(NAME a): testcode:1:4&#xa;var a = function() { return this; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:8
var a = function() { return this; }
  Parent(NAME a): testcode:1:4
var a = function() { return this; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction3(CheckGlobalThisTest.java:76)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction4" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:8&#xa;var a = function() { this.foo.bar = 6; }&#xa;  Parent(NAME a): testcode:1:4&#xa;var a = function() { this.foo.bar = 6; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:8
var a = function() { this.foo.bar = 6; }
  Parent(NAME a): testcode:1:4
var a = function() { this.foo.bar = 6; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction4(CheckGlobalThisTest.java:80)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction5" time="0.005">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION a): testcode:1:0&#xa;function a() { return function() { return this; } }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function a() { return function() { return this; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION a): testcode:1:0
function a() { return function() { return this; } }
  Parent(SCRIPT): testcode:1:0
function a() { return function() { return this; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction5(CheckGlobalThisTest.java:84)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction6" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION a): testcode:1:0&#xa;function a() { return function() { this.x = 8; } }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function a() { return function() { this.x = 8; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION a): testcode:1:0
function a() { return function() { this.x = 8; } }
  Parent(SCRIPT): testcode:1:0
function a() { return function() { this.x = 8; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction6(CheckGlobalThisTest.java:88)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction7" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:8&#xa;var a = function() { return function() { this.x = 8; } }&#xa;  Parent(NAME a): testcode:1:4&#xa;var a = function() { return function() { this.x = 8; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:8
var a = function() { return function() { this.x = 8; } }
  Parent(NAME a): testcode:1:4
var a = function() { return function() { this.x = 8; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction7(CheckGlobalThisTest.java:92)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunction8" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:8&#xa;var a = function() { return this.foo; };&#xa;  Parent(NAME a): testcode:1:4&#xa;var a = function() { return this.foo; };&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:8
var a = function() { return this.foo; };
  Parent(NAME a): testcode:1:4
var a = function() { return this.foo; };

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunction8(CheckGlobalThisTest.java:96)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testConstructor1" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testConstructor2" time="0.003" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testConstructor3" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testInterface1" time="0.004" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testOverride1" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testThisJSDoc1" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testThisJSDoc2" time="0.002" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testThisJSDoc3" time="0.004" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testThisJSDoc4" time="0.003" />
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testThisJSDoc5" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION a): testcode:1:0&#xa;function a() { /** @this x */function f() { this.foo = 56; } }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function a() { /** @this x */function f() { this.foo = 56; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION a): testcode:1:0
function a() { /** @this x */function f() { this.foo = 56; } }
  Parent(SCRIPT): testcode:1:0
function a() { /** @this x */function f() { this.foo = 56; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testThisJSDoc5(CheckGlobalThisTest.java:138)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testMethod1" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:17&#xa;A.prototype.m1 = function() { this.m2 = 5; }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;A.prototype.m1 = function() { this.m2 = 5; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:17
A.prototype.m1 = function() { this.m2 = 5; }
  Parent(ASSIGN): testcode:1:0
A.prototype.m1 = function() { this.m2 = 5; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testMethod1(CheckGlobalThisTest.java:142)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testMethod2" time="0.001">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:19&#xa;a.B.prototype.m1 = function() { this.m2 = 5; }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;a.B.prototype.m1 = function() { this.m2 = 5; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:19
a.B.prototype.m1 = function() { this.m2 = 5; }
  Parent(ASSIGN): testcode:1:0
a.B.prototype.m1 = function() { this.m2 = 5; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testMethod2(CheckGlobalThisTest.java:146)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testMethod3" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:23&#xa;a.b.c.D.prototype.m1 = function() { this.m2 = 5; }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;a.b.c.D.prototype.m1 = function() { this.m2 = 5; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:23
a.b.c.D.prototype.m1 = function() { this.m2 = 5; }
  Parent(ASSIGN): testcode:1:0
a.b.c.D.prototype.m1 = function() { this.m2 = 5; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testMethod3(CheckGlobalThisTest.java:150)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testMethod4" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:26&#xa;a.prototype[&apos;x&apos; + &apos;y&apos;] =  function() { this.foo = 3; };&#xa;  Parent(ASSIGN): testcode:1:0&#xa;a.prototype[&apos;x&apos; + &apos;y&apos;] =  function() { this.foo = 3; };&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:26
a.prototype[&apos;x&apos; + &apos;y&apos;] =  function() { this.foo = 3; };
  Parent(ASSIGN): testcode:1:0
a.prototype[&apos;x&apos; + &apos;y&apos;] =  function() { this.foo = 3; };

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testMethod4(CheckGlobalThisTest.java:154)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testPropertyOfMethod" time="0.02">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:37&#xa;a.protoype.b = {}; a.prototype.b.c = function() { this.foo = 3; };&#xa;  Parent(ASSIGN): testcode:1:19&#xa;a.protoype.b = {}; a.prototype.b.c = function() { this.foo = 3; };&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:37
a.protoype.b = {}; a.prototype.b.c = function() { this.foo = 3; };
  Parent(ASSIGN): testcode:1:19
a.protoype.b = {}; a.prototype.b.c = function() { this.foo = 3; };

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testPropertyOfMethod(CheckGlobalThisTest.java:158)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticMethod1" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:6&#xa;a.b = function() { this.m2 = 5; }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;a.b = function() { this.m2 = 5; }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:6
a.b = function() { this.m2 = 5; }
  Parent(ASSIGN): testcode:1:0
a.b = function() { this.m2 = 5; }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticMethod1(CheckGlobalThisTest.java:163)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticMethod2" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:6&#xa;a.b = function() { return function() { this.m2 = 5; } }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;a.b = function() { return function() { this.m2 = 5; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:6
a.b = function() { return function() { this.m2 = 5; } }
  Parent(ASSIGN): testcode:1:0
a.b = function() { return function() { this.m2 = 5; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticMethod2(CheckGlobalThisTest.java:167)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticMethod3" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:8&#xa;a.b.c = function() { return function() { this.m2 = 5; } }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;a.b.c = function() { return function() { this.m2 = 5; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:8
a.b.c = function() { return function() { this.m2 = 5; } }
  Parent(ASSIGN): testcode:1:0
a.b.c = function() { return function() { this.m2 = 5; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticMethod3(CheckGlobalThisTest.java:171)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testMethodInStaticFunction" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION f): testcode:1:0&#xa;function f() { A.prototype.m1 = function() { this.m2 = 5; } }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function f() { A.prototype.m1 = function() { this.m2 = 5; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION f): testcode:1:0
function f() { A.prototype.m1 = function() { this.m2 = 5; } }
  Parent(SCRIPT): testcode:1:0
function f() { A.prototype.m1 = function() { this.m2 = 5; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testMethodInStaticFunction(CheckGlobalThisTest.java:175)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunctionInMethod1" time="0.001">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:17&#xa;A.prototype.m1 = function() { function me() { this.m2 = 5; } }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;A.prototype.m1 = function() { function me() { this.m2 = 5; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:17
A.prototype.m1 = function() { function me() { this.m2 = 5; } }
  Parent(ASSIGN): testcode:1:0
A.prototype.m1 = function() { function me() { this.m2 = 5; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunctionInMethod1(CheckGlobalThisTest.java:179)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testStaticFunctionInMethod2" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:17&#xa;A.prototype.m1 = function() {  function me() {    function myself() {      function andI() { this.m2 = 5; } } } }&#xa;  Parent(ASSIGN): testcode:1:0&#xa;A.prototype.m1 = function() {  function me() {    function myself() {      function andI() { this.m2 = 5; } } } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:17
A.prototype.m1 = function() {  function me() {    function myself() {      function andI() { this.m2 = 5; } } } }
  Parent(ASSIGN): testcode:1:0
A.prototype.m1 = function() {  function me() {    function myself() {      function andI() { this.m2 = 5; } } } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testStaticFunctionInMethod2(CheckGlobalThisTest.java:183)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testInnerFunction1" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION f): testcode:1:0&#xa;function f() { function g() { return this.x; } }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function f() { function g() { return this.x; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION f): testcode:1:0
function f() { function g() { return this.x; } }
  Parent(SCRIPT): testcode:1:0
function f() { function g() { return this.x; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testInnerFunction1(CheckGlobalThisTest.java:190)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testInnerFunction2" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION f): testcode:1:0&#xa;function f() { var g = function() { return this.x; } }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function f() { var g = function() { return this.x; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION f): testcode:1:0
function f() { var g = function() { return this.x; } }
  Parent(SCRIPT): testcode:1:0
function f() { var g = function() { return this.x; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testInnerFunction2(CheckGlobalThisTest.java:194)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testInnerFunction3" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION f): testcode:1:0&#xa;function f() { var x = {}; x.y = function() { return this.x; } }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function f() { var x = {}; x.y = function() { return this.x; } }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION f): testcode:1:0
function f() { var x = {}; x.y = function() { return this.x; } }
  Parent(SCRIPT): testcode:1:0
function f() { var x = {}; x.y = function() { return this.x; } }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testInnerFunction3(CheckGlobalThisTest.java:198)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testInnerFunction4" time="0.001">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION f): testcode:1:0&#xa;function f() { var x = {}; x.y(function() { return this.x; }); }&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function f() { var x = {}; x.y(function() { return this.x; }); }&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION f): testcode:1:0
function f() { var x = {}; x.y(function() { return this.x; }); }
  Parent(SCRIPT): testcode:1:0
function f() { var x = {}; x.y(function() { return this.x; }); }

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testInnerFunction4(CheckGlobalThisTest.java:203)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testIssue182a" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:16&#xa;var NS = {read: function() { return this.foo; }};&#xa;  Parent(STRING_KEY read): testcode:1:10&#xa;var NS = {read: function() { return this.foo; }};&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:16
var NS = {read: function() { return this.foo; }};
  Parent(STRING_KEY read): testcode:1:10
var NS = {read: function() { return this.foo; }};

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testIssue182a(CheckGlobalThisTest.java:208)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testIssue182b" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:17&#xa;var NS = {write: function() { this.foo = 3; }};&#xa;  Parent(STRING_KEY write): testcode:1:10&#xa;var NS = {write: function() { this.foo = 3; }};&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:17
var NS = {write: function() { this.foo = 3; }};
  Parent(STRING_KEY write): testcode:1:10
var NS = {write: function() { this.foo = 3; }};

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testIssue182b(CheckGlobalThisTest.java:212)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testIssue182c" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:25&#xa;var NS = {}; NS.write2 = function() { this.foo = 3; };&#xa;  Parent(ASSIGN): testcode:1:13&#xa;var NS = {}; NS.write2 = function() { this.foo = 3; };&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:25
var NS = {}; NS.write2 = function() { this.foo = 3; };
  Parent(ASSIGN): testcode:1:13
var NS = {}; NS.write2 = function() { this.foo = 3; };

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testIssue182c(CheckGlobalThisTest.java:216)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testIssue182d" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION Foo): testcode:1:0&#xa;function Foo() {} Foo.prototype = {write: function() { this.foo = 3; }};&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function Foo() {} Foo.prototype = {write: function() { this.foo = 3; }};&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION Foo): testcode:1:0
function Foo() {} Foo.prototype = {write: function() { this.foo = 3; }};
  Parent(SCRIPT): testcode:1:0
function Foo() {} Foo.prototype = {write: function() { this.foo = 3; }};

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testIssue182d(CheckGlobalThisTest.java:220)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testLendsAnnotation1" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:57&#xa;/** @constructor */ function F() {}dojo.declare(F, {foo: function() { return this.foo; }});&#xa;  Parent(STRING_KEY foo): testcode:1:52&#xa;/** @constructor */ function F() {}dojo.declare(F, {foo: function() { return this.foo; }});&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:57
/** @constructor */ function F() {}dojo.declare(F, {foo: function() { return this.foo; }});
  Parent(STRING_KEY foo): testcode:1:52
/** @constructor */ function F() {}dojo.declare(F, {foo: function() { return this.foo; }});

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testLendsAnnotation1(CheckGlobalThisTest.java:225)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testLendsAnnotation2" time="0.003">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:84&#xa;/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.bar} */ (    {foo: function() { return this.foo; }}));&#xa;  Parent(STRING_KEY foo): testcode:1:79&#xa;/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.bar} */ (    {foo: function() { return this.foo; }}));&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:84
/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.bar} */ (    {foo: function() { return this.foo; }}));
  Parent(STRING_KEY foo): testcode:1:79
/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.bar} */ (    {foo: function() { return this.foo; }}));

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testLendsAnnotation2(CheckGlobalThisTest.java:230)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testLendsAnnotation3" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:90&#xa;/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.prototype} */ (    {foo: function() { return this.foo; }}));&#xa;  Parent(STRING_KEY foo): testcode:1:85&#xa;/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.prototype} */ (    {foo: function() { return this.foo; }}));&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:90
/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.prototype} */ (    {foo: function() { return this.foo; }}));
  Parent(STRING_KEY foo): testcode:1:85
/** @constructor */ function F() {}dojo.declare(F, /** @lends {F.prototype} */ (    {foo: function() { return this.foo; }}));

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:408)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:741)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testLendsAnnotation3(CheckGlobalThisTest.java:236)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testSuppressWarning" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION ): testcode:1:8&#xa;var x = function() { this.complex = 5; };&#xa;  Parent(NAME x): testcode:1:4&#xa;var x = function() { this.complex = 5; };&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION ): testcode:1:8
var x = function() { this.complex = 5; };
  Parent(NAME x): testcode:1:4
var x = function() { this.complex = 5; };

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testSuppressWarning(CheckGlobalThisTest.java:242)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testArrowFunction1" time="0.004">
    <failure message="There should be one warning, repeated 2 time(s). Warnings: [] expected:&lt;2&gt; but was:&lt;0&gt;" type="junit.framework.AssertionFailedError">junit.framework.AssertionFailedError: There should be one warning, repeated 2 time(s). Warnings: [] expected:&lt;2&gt; but was:&lt;0&gt;
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1098)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testArrowFunction1(CheckGlobalThisTest.java:247)
</failure>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testArrowFunction2" time="0.002">
    <failure message="There should be one warning, repeated 2 time(s). Warnings: [] expected:&lt;2&gt; but was:&lt;0&gt;" type="junit.framework.AssertionFailedError">junit.framework.AssertionFailedError: There should be one warning, repeated 2 time(s). Warnings: [] expected:&lt;2&gt; but was:&lt;0&gt;
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1098)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testArrowFunction2(CheckGlobalThisTest.java:252)
</failure>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testArrowFunction3" time="0.001">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION Foo): testcode:1:0&#xa;function Foo() {} Foo.prototype.getFoo = () =&gt; this.foo;&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function Foo() {} Foo.prototype.getFoo = () =&gt; this.foo;&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION Foo): testcode:1:0
function Foo() {} Foo.prototype.getFoo = () =&gt; this.foo;
  Parent(SCRIPT): testcode:1:0
function Foo() {} Foo.prototype.getFoo = () =&gt; this.foo;

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testArrowFunction3(CheckGlobalThisTest.java:257)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <testcase classname="com.google.javascript.jscomp.CheckGlobalThisTest" name="testArrowFunction4" time="0.002">
    <error message="INTERNAL COMPILER ERROR.&#xa;Please report this problem.&#xa;&#xa;null&#xa;  Node(FUNCTION Foo): testcode:1:0&#xa;function Foo() {} Foo.prototype.setFoo = (f) =&gt; { this.foo = f; };&#xa;  Parent(SCRIPT): testcode:1:0&#xa;function Foo() {} Foo.prototype.setFoo = (f) =&gt; { this.foo = f; };&#xa;" type="java.lang.RuntimeException">java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(FUNCTION Foo): testcode:1:0
function Foo() {} Foo.prototype.setFoo = (f) =&gt; { this.foo = f; };
  Parent(SCRIPT): testcode:1:0
function Foo() {} Foo.prototype.setFoo = (f) =&gt; { this.foo = f; };

	at com.google.javascript.jscomp.CheckGlobalThis.shouldTraverse(CheckGlobalThis.java:94)
	at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
	at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:165)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:557)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:570)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:289)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:528)
	at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:1044)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:935)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:543)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:509)
	at com.google.javascript.jscomp.CompilerTestCase.test(CompilerTestCase.java:450)
	at com.google.javascript.jscomp.CompilerTestCase.testSame(CompilerTestCase.java:752)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testFailure(CheckGlobalThisTest.java:36)
	at com.google.javascript.jscomp.CheckGlobalThisTest.testArrowFunction4(CheckGlobalThisTest.java:263)
Caused by: java.lang.NullPointerException
</error>
  </testcase>
  <system-out><![CDATA[]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
